src/main/java/cinema/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package cinema.service.impl;
+
+import cinema.dao.ShoppingCartDao;
+import cinema.dao.TicketDao;
+import cinema.injections.Inject;
+import cinema.injections.Service;
+import cinema.model.MovieSession;
+import cinema.model.ShoppingCart;
+import cinema.model.Ticket;
+import cinema.model.User;
+import cinema.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket(movieSession, user);
+        ticketDao.add(ticket);
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user);
+        shoppingCart.getTickets().add(ticket);
+        shoppingCartDao.update(shoppingCart);
```

Comment:
I'm not sure about the order, because you add a ticket, but can catch the exception in `getByUser(user)` or `update(shoppingCart)`

Reply:
Ideally, creating a ticket and associating it with the shopping cart should be a single transaction, but we are obliged to use separate DAOs